enterprise/views.py

```diff
@@ -291,23 +291,95 @@ def get_course_specific_consent(self, request, course_id):
             'policy_link_template': '',
         }
         context_data.update(course_specific_context)
-        if customer.require_account_level_consent:
-            context_data.update({
-                'consent_request_prompt': _(
-                    'To access this and other courses sponsored by {bold_start}{enterprise_customer_name}{bold_end}, '
-                    'and to use the discounts available to you, you must first consent to share your '
-                    'learning achievements with {bold_start}{enterprise_customer_name}{bold_end}.'
-                ).format(
-                    enterprise_customer_name=customer.name,
-                    bold_start='<b>',
-                    bold_end='</b>',
-                ),
-                'requested_permissions': [
-                    _('your enrollment in all sponsored courses'),
-                    _('your learning progress'),
-                    _('course completion'),
-                ],
-            })
+
+        return render(request, 'enterprise/grant_data_sharing_permissions.html', context=context_data)
+
+    @method_decorator(login_required)
+    def get_program_specific_consent(self, request, program_uuid):
```

Comment:
it looks like this has a lot in common with the course specific function, is there more opportunity to consolidate, particularly with the context? although based on the conversation in this pr it sounds like something you plan to address later when cleaning up the course specific logic.

Reply:
Yeah, cleanup/consolidation will come later.